sdk/src/main/java/io/horizen/account/api/rpc/types/TransactionArgs.java

```diff
@@ -63,54 +108,45 @@ public EthereumTransaction toTransaction(NetworkParams params) throws RpcExcepti
                 String.format("invalid chainID: got %d, want %d", chainId, saneChainId)
             ));
         }
+        if (nonce == null) {
+            // TODO: should automatically use current nonce from "latest" state
```

Comment:
the `TODOs` in this method are not very clear to me.
Do we need this to be changed now? how critical is it?

Reply:
I added the Todos while going through the equivalent implementation in GETH.

I don't think this is critical at all because it is only relevant when the node is signing a transaction, i.e. `eth_signTransaction` or `eth_sendTransaction`, which no enduser will ever use. Also, most tools will fill all these parameters anyway so we don't need to auto-fill them, e.g. a tool will likely call `eth_estimateGas` itself and then put the `gas` value, instead of omitting `gas` and expecting the node to estimate gas usage.

Reply:
I just added the checks and exceptions to prevent `NullPointerExceptions` down the road, because e.g. `EthereumTransaction.gasLimit` is `null`. The deployment issue in YUMA currently is caused by this.

Reply:
thanks, now it is clear.